Podcast Episode page setup

rccsousa opened the pull request:
Why:
* Setup podcast episode page to display content

How:
* create multiple component blocks to encapsulate each section

rccsousa committed:
Create pages for collections and graphQL schema

- scaffolding pages for new collections
- graphQL schema generation
- updating payload-types.ts for content fetching

rccsousa committed:
Setup Podcast Episode pipeline

- generated GraphQL schemas;
- created fetchers;
- created basic landing page for testing slug navigation;

rccsousa committed:
update podcast-episode pipeline and new hooks

- created contributor hook to populate contributor field on read,
- updated graphQL queries

rccsousa committed:
Setup podcast episode pages

- create custom components for each section

rccsousa committed:
Fix Contributors access control

Contributors are now publicly available.

Updated podcastEpisodes graphQL queries accordingly.

rccsousa committed:
WIP: episode head block setup

rccsousa committed:
wip: update headblock

- created methods to fetch episode duration and respective helper
  functions

rccsousa committed:
fix: update PodcastEpisodes collection

- deleted unnecessary populateContributors method,
- moved back to 'collections' folder root for simplicity,

rccsousa committed:
setup podcast-episodes pages

- created blocks for each section of the page
- created helper functions to populate different fields

rccsousa committed:
fix: fixed EpisodeHead placeholder styling

rccsousa committed:
Merge branch 'main' into rs/podcast-page-setup

davelange commented on src/app/_blocks/EpisodeContent/index.tsx:
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import { PodcastEpisode } from "@/payload/payload-types";
+
+export const EpisodeContent: React.FC<{ episode: PodcastEpisode }> = ({ episode }) => {
```
Just a heads up that the React.FC type isnt totally necessary. You can just do 
``` 
const EpisodeContent = ({ episode }: { episode: PodcastEpisode }) => ...
```
But it's up to you 👍

davelange commented on src/payload/payload.config.ts:
```diff
@@ -19,7 +19,7 @@ import { Posts } from "./collections/Posts";
 import { Contributors } from "./collections/Contributors";
 import Users from "./collections/Users";
-import { PodcastEpisodes } from "./collections/PodcastEpisodes";
+import { PodcastEpisodes } from "./collections/PodcasstEpisodes";
```
`PodcasstEpisodes`? 😂

rccsousa replied:
Inspired by "snake jazz" 😆 👉 👈 

i've fixed it in the other branch (just noticed yesterday) , will push it to here as well

rccsousa replied:
fixed ✅

davelange commented on src/app/_blocks/EpisodeHead/index.tsx:
```diff
@@ -0,0 +1,57 @@
+        <div>
+          <ContentTypePill />
+          <h5>{title}</h5>
+          <h6>episode.podcastName property</h6>
```
Is this meant to be `<h6>{episode.podcastName}</h6>`?

rccsousa replied:
based on Figma, yes. 


i've added an `h6` for now for refference but it will be formatted accordingly when we move on to styling

davelange commented on src/app/(pages)/podcast-episodes/[slug]/page.tsx:
```diff
@@ -1,30 +1,27 @@
+"use client"
```
Why is "use client" on this page? This is causing some errors fpr me because async/await is only supported in server components, not client

rccsousa replied:
it was there when i was constructing the components, fixing it now

rccsousa replied:
fixed ✅

davelange commented on src/app/_utilities/getEpisodeDuration.ts:
```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+import { formatEpisodeDuration } from "@/app/_utilities/formatEpisodeDuration";
+
+export function getEpisodeDuration(src) {
```
Because this functions uses hooks (useState, useEffect), it's technically also a hook. It's common practice to call hooks useSomething, so in this case `useEpisodeDuration`

rccsousa replied:
fixed ✅

davelange commented on src/app/_utilities/formatDateTime.ts:
```diff
@@ -1,20 +1,20 @@
-  // const MinMin = (minutes < 10) ? `0${minutes}` : minutes;
-  // const SS = (seconds < 10) ? `0${seconds}` : seconds;
+  // Array of full month names
+  const monthNames = [
```
It would be easier to use something like this:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat
Feel free to change this in a future PR though, it's ok for now

davelange commented on src/app/_utilities/formatDateTime.ts:
```diff
@@ -1,20 +1,20 @@
+  const YYYY = date.getFullYear();
+
+  const monthName = monthNames[months]; // Get the full name of the month
+  const DD = days < 10 ? `0${days}` : days; // Add leading zero to day if needed
```
FYI, you can use padStart for this 😉   https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

rccsousa replied:
great! thanks

rccsousa committed:
fix: typos and renamings

- fixed "PodcasstEpisodes" typo;
- renamed getEpisodeDuration to useEpisodeDuration;
